src/common/pjrt_implementation/loaded_executable_instance.cc

```
@@ -76,65 +86,196 @@ LoadedExecutableInstance::Execute(PJRT_LoadedExecutable_Execute_Args *args) {
-    auto result_buffer = std::make_unique<BufferInstance>(
-        *this->addressable_devices_[dev_index], untilized_output_tensor,
-        image_->get_output_shape(i), image_->get_output_stride(i));
+  for (size_t i = 0; i < rt_outputs.size(); ++i) {
     tt::runtime::deallocateTensor(rt_outputs[i], /*force=*/true);
-
-    result_buffer->setType(image_->get_output_types()[i]);
-
-    DLOG_F(INFO, "Runtime output id: %d", result_buffer->unique_id());
-    args->output_lists[dev_index][i] = *(result_buffer.release());
   }
 
   if (args->device_complete_events) {
-    args->device_complete_events[dev_index] = *(new EventInstance());
+    for (int i = 0; i < num_devices; i++)
+      args->device_complete_events[i] = *(new EventInstance());
   }
 
   tt::runtime::closeDevice(device);
 
   return tt_pjrt_status::kSuccess;
 }
 
+std::unordered_map<std::string, std::string>
+LoadedExecutableInstance::getStrategyMapFromSharding(
+    const mlir::tt::sharding_utils::MeshSharding &meshSharding,
+    size_t num_devices) {
+  mlir::tt::MeshShardType meshType = meshSharding.getShardType();
+  std::unordered_map<std::string, std::string> strategy;
+  if (meshType == mlir::tt::MeshShardType::Replicate) {
+    if (num_devices == 1) {
+      strategy["strategy"] = "identity";
+    } else {
+      strategy["strategy"] = "replicate";
+      strategy["replication_factor"] = std::to_string(num_devices);
+    }
+  } else if (meshType == mlir::tt::MeshShardType::Devices) {
+    llvm::ArrayRef<int64_t> shardShape = meshSharding.getShardShape();
+    if (shardShape.size() == 2) {
+      strategy["strategy"] = "shard_2d";
+      strategy["mesh_shape_y"] = std::to_string(shardShape[0]);
+      strategy["mesh_shape_x"] = std::to_string(shardShape[1]);
+    } else if (shardShape.size() == 1) {
+      strategy["strategy"] = "shard";
+      strategy["shard_dim"] = "0";
+    } else {
+      DLOG_F(ERROR, "Invalid mesh sharding type");
+      return strategy;
```

If you transfer this function into tt-mlir, this should throw an error. Below too. Otherwise, we would return `tt_pjrt_status::kInternal` from here.